### PR TITLE
MVP v5 docs

### PR DESCRIPTION
**Pull request by davidkpiano**

### CODE REVIEW

**Review by laurakalbag — APPROVED** [review]
Fabulous!

**Comment by laurakalbag** [comment]
Just merged latest main in.

File: versioned_docs/version-5/migration.mdx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+`actor.send(eventObject)` no longer accepts a string; must pass full event object `{ type: '...' }`
+
+No more null events; use `always`
+
+2nd argument to `createMachine` is deprecated: `createMachine(config, options)` is now `createMachine(config).provide(options)`
+
+No more `.withConfig()`; use `.provide()` instead
+
+No more `.withContext()`; use input instead (see below)
```
Comment (with-heart) [inline versioned_docs/version-5/migration.mdx:31]: There's no below to see!

Comment (davidkpiano) [reply versioned_docs/version-5/migration.mdx:31]: WIP! 😅

Comment (with-heart) [reply versioned_docs/version-5/migration.mdx]: Of course! 😁

Would it be helpful to note things like that and also things that I notice are missing, or do you have it pretty well covered?

Comment (davidkpiano) [reply versioned_docs/version-5/migration.mdx]: Yes, that'd be helpful 🙏

Comment (with-heart) [reply versioned_docs/version-5/migration.mdx]: Oki will do!

File: versioned_docs/version-5/migration.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+`state.history` is removed
+
+Batched events `actor.batch([...])` is removed
+
+`config.schema` is now `config.types`
+
+`config.tsTypes` is now in `config.types.typegen`
+
+No more `in: '...'` guards; use `stateIn(...)` instead
+
```
Comment (with-heart) [inline versioned_docs/version-5/migration.mdx:46]:
```suggestion

Properties on action and guard objects other than `type` should be nested under a `params` object; `{ type: 'someType', message: 'hello' }` becomes `{ type: 'someType', params: { message: 'hello' }}`

```

File: versioned_docs/version-5/migration.mdx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+---
+title: 'Migrating from XState v4 to v5'
+---
+
+### Breaking changes
+
+`Machine` to `createMachine`
+
+`cond` is now `guard`
```
Comment (with-heart) [inline versioned_docs/version-5/migration.mdx:9]: I think we could make this more of a guide than a list of changes if we flesh each item out a little bit. I was thinking something like this:

````suggestion
#### `Machine` is now `createMachine`

```diff
-import { Machine } from 'xstate';
+import { createMachine } from 'xstate';

-const machine = Machine({});
+const machine = createMachine({});
```

#### `cond` is now `guard`

```diff
 createMachine({
   on: {
     event: {
-     cond: 'someGuard',
+     guard: 'someGuard',
     },
   },
 });
```
````

Make each item a heading (for generating a table of contents with links) and add a diff to make it clear with code what the change means.

I'm building diffs for all of the changes for https://github.com/statelyai/xstate-tools/issues/321 so we could reuse that here as well.

Comment (davidkpiano) [reply versioned_docs/version-5/migration.mdx:9]: Yeah, while this is in draft, I'm just keeping the items we need to expand as "bullet points" (notes to self).

**Pull request merged**